woocommerce/google-listings-and-ads
# Show validation errors on step 2 of the onboarding flow when unable to continue

### Changes proposed in this Pull Request:


This is a part of implementation for #634 and for 📌 [Display form validation errors](https://github.com/woocommerce/google-listings-and-ads/issues/1993#tasks-display-validation-errors) in #1993.

To show validation errors on step 2 of the onboarding flow when unable to continue, this PR:

#### Extract the shareable component and adjust the layouts of shipping time components

- Add a new component `ValidationErrors` for showing form validation error messages, and replace the errors rendering in `AssetGroupCard` with it.
- Move the Card layouts from `ShippingTimeSection` into `ShippingTimeSetup` and remove unneeded codes.

#### Fix and adjust validator `js/src/components/free-listings/configure-product-listings/checkErrors.js`

- Fix incorrect keys and rephrase a few messages.
- Update the validation of tax rate data to check if the store country code is `US`.

#### Add the mechanism of showing validation errors to the product listings form

- Request showing validation errors when clicking the submit of product listings form if any data is invalid.
- Add a function to render validation errors to the product listings form.
- Add the rendering of validation errors to:
   - `ChooseAudienceSection`
   - `FlatShippingRatesInputCards`
   - `EstimatedShippingRatesCard`
   - `OfferFreeShippingCard`
   - `MinimumOrderCard`
   - `ShippingTimeSetup`
   - `TaxRate`

### Screenshots:

https://github.com/woocommerce/google-listings-and-ads/assets/17420811/4c977398-fb9b-4abc-9ef6-fc551cca362b

### Detailed test instructions:

Some free listings data are unable to clean via UI. It might need to run the following SQL to clean them from database.

```sql
DELETE FROM wp_options WHERE option_name = 'gla_target_audience';
DELETE FROM wp_options WHERE option_name = 'gla_merchant_center';
```

1. Go to step 2 of the onboarding flow.
2. Before clicking the "Continue" button, the UI should not show any validation errors even if there are invalid data.
3. Check if the "Continue" button is always enabled no matter whether the data are valid or not.
4. With some invalid data, click the "Continue" button to see if the validation errors are shown in the corresponding places.
5. Change data to view if the validation errors are shown or hidden accordingly.
6. Make all data valid and click the "Continue" button to check if it goes to step 3.

### Additional details:

💡 The implementation for step 3 will be done by subsequent PR(s)

### Changelog entry
